app/domain/authentication/authn_jwt/authenticator.rb

```diff
@@ -6,39 +6,47 @@ module AuthnJwt
     # request is valid, and return conjur authn token accordingly
     Authenticate = CommandClass.new(
       dependencies: {
-        token_factory: TokenFactory.new
+        token_factory: TokenFactory.new,
+        logger: Rails.logger
       },
       inputs: %i[jwt_configuration authenticator_input]
     ) do
 
       def call
+        create_authentication_parameters
         validate_and_decode_token
-        conjur_id
+        get_jwt_identity
         validate_restrictions
         new_token
       end
 
       private
 
+      def create_authentication_parameters
+        @logger.debug(LogMessages::Authentication::AuthnJwt::CREATING_AUTHENTICATION_PARAMETERS_OBJECT.new)
+        @authentication_parameters = Authentication::AuthnJwt::AuthenticationParameters.new(@authenticator_input)
+      end
+
       def validate_and_decode_token
-        #TODO : Need to decide how the token is going to be given and parsed and change this line according to this
-        unless @jwt_configuration.validate_and_decode_token(jwt_token: @authenticator_input.credentials)
-          raise "validate and decode failed"
-        end
+        @logger.debug(LogMessages::Authentication::AuthnJwt::VALIDATE_AND_DECODE_TOKEN.new)
+        @authentication_parameters.decoded_token = @jwt_configuration.validate_and_decode_token(@authentication_parameters)
       end
 
-      def conjur_id
+      def get_jwt_identity
         # Will be changed when real get identity implemented
-        @jwt_configuration.conjur_id
+        @logger.debug(LogMessages::Authentication::AuthnJwt::GET_JWT_IDENTITY.new)
+        @authentication_parameters.jwt_identity =  @jwt_configuration.jwt_identity(@authentication_parameters)
       end
 
       def validate_restrictions
-        unless @jwt_configuration.validate_restrictions
+        @logger.debug(LogMessages::Authentication::AuthnJwt::CHECKING_IDENTITY_FIELD_EXISTS.new)
+        unless @jwt_configuration.validate_restrictions(@authentication_parameters)
```

Comment:
Please move this to an error class ` " raise "not matching policy annotations"`

Reply:
If it not related to your changes we will do it in another PR

Reply:
Will do in another PR